### ISSUE TITLE
Add Lint Capabilities to the dbtcloud_job resource 

### ISSUE THREAD

**Issue by SteveFrensch** [dbt-labs/terraform-provider-dbtcloud#310]
## Is your feature request related to a problem? Please describe.

There is a new DBT Cloud feature where sqlfluff linting that can be added to CI jobs (https://docs.getdbt.com/docs/deploy/continuous-integration#sql-linting) 

They have updated the `create_job` [API](https://docs.getdbt.com/dbt-cloud/api-v2#/operations/Create%20Job) with two new boolean fields to manage this:

```
  "run_lint": true,
  "errors_on_lint_failure": true,
```
It would be great if the `dbtcloud_job` resource was able to mange these fields. 

## Describe the solution you'd like
Maybe we can add the fields to the resource like:

```
  run_lint: true,
  errors_on_lint_failure: true,
```

i.e.

```
resource "dbtcloud_job" "sample CI Job" {
  environment_id = dbtcloud_environment.ci_environment.environment_id
   execute_steps = [
    "dbt build --warn-error -s state:modified+ --fail-fast"
  ]
  generate_docs        = false
  name                 = "CI JOB"
  num_threads          = 32
  project_id           = dbtcloud_project.data_engineering_project.id
  run_generate_sources = false
  run_lint           = true,
  errors_on_lint_failure   =  true,
  target_name          = "prod"
   ...
}
```

**Comment by SteveFrensch** [comment]
Just noticed this is still in beta and not yet widely available, so may be premature for inclusion.

**Comment by b-per** [comment]
I can do some testing. If I can add it to TF without breaking it for users who don't have access to the Beta, I might still be able to do it.

**Comment by tturrisi-accelins** [comment]
Hi, do you know when this feature might be implemented? We're currently using this flag in dbt and would love to be able to get the config into our TF. We're currently having to set the flag manually in the UI and watch carefully to make sure other releases don't override it.

**Comment by csquire** [comment]
Hey all, I'll have a look at this and see if we can get it added.

**Comment by sigveks** [comment]
When will this be deployed and available?

**Comment by b-per** [comment]
Hi @sigveks ! The team released 0.3.23 yesterday, so it is available now.

**Comment by sigveks** [comment]
@b-per I see! So there are routinely deployments that do not updated the documentation site?
